src/H5Shyper.c

```diff
@@ -3229,25 +3231,38 @@ H5S__hyper_copy(H5S_t *dst, const H5S_t *src, bool share_selection)
 
     /* Check if there is hyperslab span information to copy */
     /* (Regular hyperslab information is copied with the selection structure) */
-    if (src->select.sel_info.hslab->span_lst != NULL) {
+    if (src_hslab->span_lst != NULL) {
         if (share_selection) {
             /* Share the source's span tree by incrementing the reference count on it */
-            dst->select.sel_info.hslab->span_lst = src->select.sel_info.hslab->span_lst;
-            dst->select.sel_info.hslab->span_lst->count++;
+            dst_hslab->span_lst = src_hslab->span_lst;
+            dst_hslab->span_lst->count++;
         } /* end if */
-        else
+        else {
             /* Copy the hyperslab span information */
-            dst->select.sel_info.hslab->span_lst =
-                H5S__hyper_copy_span(src->select.sel_info.hslab->span_lst, src->extent.rank);
```

Comment:
Code wasn't previously checking for errors here, which would hide problems when trying to copy a dataspace with an extent of 0 dims.